src/Apps/Components/AppShell.tsx

```diff
@@ -27,6 +28,7 @@ export const AppShell: React.FC<AppShellProps> = props => {
   useRunAuthIntent()
   useAuthValidation()
   useSocialAuthTracking()
+  useCountryCode()
```

Comment:
More auth stuff 😛 Mentioned this in another PR, but can we roll all of these hooks into a single `useAuth` hook?

Reply:
I honestly don't really buy the argument to do this but I'm not opposed to it.

Reply:
Will just amend this PR

Reply:
There were two auth hooks. then there was another auth related hook. and now another. Clearly, auth is a thing that grows and changes and thus should have a canonical source of reference for all setup-related things! Its general importance as a concept also warrants it. A single file that rolls it all up. Thanks for the update 👍

Reply:
Yeah OK, I'm convinced

Reply:
Really struggling with the naming here: it's not any kind of setup and all of these things are only loosely related.

`useRunAuthIntent`: picks up any after auth actions that were set before authentication and runs them.
`useAuthValidation`: just checks to see if your current authentication is still valid and logs you out if it's not.
`useSocialAuthTracking`: picks up after social authentication to track a successful login or sign up
`useCountryCode`: pre-fetches the country code for the sign up form.

So one something that happens "before"; the others are things that happen "after".

`useAuthEnhancements`?

Reply:
I'd say we just keep it simple with the name. `useSetupAuth` is general enough to contain loosely grouped things. Its also signal enough to add future things to that spot.